sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsException.cs

```diff
@@ -216,7 +217,13 @@ public enum FailureReason
             ServiceTimeout,
 
             /// <summary>There was a general communications error encountered when interacting with the Azure Event Hubs service.</summary>
-            ServiceCommunicationProblem
+            ServiceCommunicationProblem,
+
+            /// <summary>A client was forcefully disconnected from an Event Hub instance.  This typically occurs when another consumer with higher <see cref="PartitionPublishingOptions.OwnerLevel" /> asserts ownership over the partition and producer group.</summary>
+            ProducerDisconnected,
+
+            /// <summary>A client is in an invalid state from which it cannot recover.  It is recommended that the client be closed and recreated to force reinitialization of state.</summary>
+            InvalidClientState
```

Comment:
So would this fall under a "this is recoverable, so the SDK should just do this internally"? Sort of like retries?

Reply:
No; this is a sign that there is a deeper problem where the service and client disagree on the state of the world.  We discussed this quite a bit and determined that we cannot deterministically prove that it is safe to recover.  

As the comment indicates, this is a signal that the client is corrupt and callers should create a new instance.

Reply:
Yeah, I read the comment and thought "yes, but....how corrupt?".